### ISSUE TITLE
Secondary sales

### ISSUE THREAD

**Issue by neverhood311** [hicetnunc2000/hicetnunc#184]
OBJKT owners should be able to sell/swap their OBJKTs, even if they're not the original creator.

**Comment by veqtor** [comment]
Yes, this is being worked on

**Comment by Quasimondo** [comment]
Now that secondary sales are working I want to use this thread to discuss a few questions and feature requests related to this topic:

Question: What are the current mechanics of secondary sales regarding the order in which they are processed? Example if one seller offers one OBJKT from an edition for 20 tez and another one offers an OBJKT from the same edition for 10 tez but enters the swap 1 hour later, which one of the two takes precedence? The lowest price or the earlier swap? In my opinion it should always be the lowest available offer since otherwise people could block the market of an OBJKT by putting in a 1000000 tez sale.

Feature Request: ideally all current secondary market offers for an OBJKT should be listed, which would allow people to get a feeling for the supply and demand of a piece.

Bug: I am sure this already in the works, but right now we are still seeing the "not for sale" label even if an edition is for sale on the secondary market

**Comment by codewithfeeling** [comment]
Ideally the creator's swaps would precede all secondary swaps regardless of price, otherwise this would effectively eliminate the option for a creator to do progressive pricing until all of their work is sold.

Say I minted 100 editions and started with an initial swap of 10 at 10tez, and gradually ramped up to 50tez in batches of 20. Say I've then sold all my 30tez editions but haven't got round to starting on my 40tez ones. The site would should "Not for sale" and the secondary market would kick in. People who bought early could sell for a profit at 30tez, and I'd be in the queue behind them trying to get my own swaps out, and my 40tez price would be out the window.

This is possibly healthy for competitive pricing but I'd be interested to hear thoughts on this.

Is it outrageous to suggest that the artist/creator can decide when to enable the secondary market on their work?

**Comment by Quasimondo** [comment]
Oh that is an interesting take. I can see your point, but I personally feel that this would go against the idea of a free market. I fear that this and the option to forbid NFTs to be resold could be detrimental to a viable market in which also speculative collectors play their role in bringing new funds to our little economy. Having too much protectionism might make potential collectors rather stay away.

**Comment by Quasimondo** [comment]
Also in the case of the staggered sales with increasing prices principle I believe that an early flipper would rather shoot themselves in the foot if they underbid you since if they waited until the whole edition is sold out their chances of making an even higher profit would be much better. On the other hand if your pieces do not sell for the high price you were hoping for it also gives you a signal that you might have to reconsider your pricing strategy until the collectors see your "true value".

**Comment by Quasimondo** [comment]
Maybe something more karma and reputation based might help to avoid unpleasant or aggressive behavior. So if someone practices "price trolling" or similar toxic behavior that wallet id could be reported and some measures like a temporary ban from trading or a three strikes rule could be imposed.

**Comment by codewithfeeling** [comment]
I’m all for that. Visibility of multiple sellers’ reputations and involvement in the community would be great. It would also neatly solve the point I raised before - if I can see that one of my options is to collect direct from the artist even if that is more expensive than picking up a cheaper one from someone with zero karma, I would do it. That still keeps the free market alive - if someone else just wants the cheapest option, they can have it.

**Comment by Quasimondo** [comment]
I just noticed that there is a pending pull request related to the topic and I believe that this means that currently really the sales queue is ordered by time (which is not ideal in my opinion as mentioned above):
https://github.com/hicetnunc2000/hicetnunc/pull/215

**Comment by codewithfeeling** [comment]
Completely agree with @Quasimondo that ordering by time is potentially extremely problematic - all it takes is one 1000000tez secondary ask and that's the end of the line for that piece.

**Comment by codewithfeeling** [comment]
I released a piece yesterday - 100 editions of a fun thing with unicorns in space to celebrate #OBJKT4OBJKT. The whole idea was to share the art for free. Now 5 editions are on the secondary market before the swap weekend is even over, and someone wants 15tez for the first of these. Secondary sales are totally locked up until this is paid.

It's like a ransom.

![image](https://user-images.githubusercontent.com/262516/112757986-f541a400-8fe3-11eb-8351-c432488c8a0a.png)

**Comment by AliaK77** [comment]
thinking out loud..

could there be a way for the original artist (OA) to put a secondary sale (SS) on hold (perhaps for a time period? or some criteria eg % of qty the OA is selling are sold) if the OA wants to sell in a batch and is blocked by a reseller trying to sell at a higher price. then once the OA sales are completed, the SS becomes active again. if the OA isn't selling at the time of the SS, then the SS proceeds/is active.

or.. (related to ^^)

I haven't had a chance to check the UI, could another option (or the way ^^ is implemented) be for the UI to show all current/active sales (& history?) for the OBJKT - then if the OA is selling at a lower price than the SS, at same time, the buyer is likely to purchase at the lower price, so then the OA can effectively 'block' the SS (ie reverse scenario)

**Comment by renatachebel** [comment]
maybe a link on the objkt page to see a list of the sellers and the prices it's being sold for? could be in the owners tab too, a column after the owners listing the price they are selling it for, listing original artist first?

**Comment by skenaja** [comment]
Most of the requests here have now been implemented.
- secondary market is active
- view prices on `collectors` tab
- lowest price out of all the swaps used for `collect for x tez` on objkt details
- prices sorted in ascending order on collectors tab

If you would like some new features that pertain to the secondary market, it's worth adding them as individual requests for tracking and discussion.
